content/docs/fundamentals/converting/resolving-errors@zh_CN.md

```diff
@@ -0,0 +1,350 @@
+
+{% call callout('注意', type='note') %}
+除了必须采用内嵌样式以外，所有样式信息的文件大小还不能超过 50 千字节的上限。您应该先使用 [SASS](http://sass-lang.com/) 等 CSS 预处理器压缩您的 CSS，然后再将 CSS 内嵌到 AMP 网页中。
+{% endcall %}
+
+{% call callout('重要提示', type='caution') %}
+整个 AMP 文档中只能有 1 个样式标记。如果 AMP 网页引用了多个外部样式表，则您需要将这些样式表整理为一组规则。要了解哪些 CSS 规则在 AMP 中有效，请参阅[受支持的 CSS](/zh_cn/docs/design/responsive/style_pages.html)。
+{% endcall %}
+
+## 排除第三方 JavaScript
+
+虽然对于 AMP 网页，您可以通过内嵌 CSS 的方式相对轻松地重新制作样式表，但 JavaScript 却并非如此。
+
+<pre class="error-text">
+不允许使用标记“script”（特定形式除外）。
+</pre>
+
+一般来说，只有符合以下两项主要要求的脚本才允许在 AMP 中使用：
+
+1.  所有 JavaScript 都必须是异步的（即脚本标记中包含 `async` 属性）。
+2.  JavaScript 用于 AMP 库以及网页中的任何 AMP 组件。
+
+这样实际上排除了在 AMP 中使用任何由用户生成的 JavaScript/第三方 JavaScript 的行为，如下所述的情况除外。
+
+{% call callout('注意', type='note') %}
+关于由用户生成的脚本/第三方脚本的限制只存在以下例外情况：
+
+1.  将元数据添加到网页或配置 AMP 组件的脚本。这些脚本将具有类型属性 `application/ld+json` 或 `application/json`。
+2.  iframe 中包含的脚本。在 iframe 中包含 JavaScript 应该是在万不得已时才使用的措施。应尽可能地将 JavaScript 功能替换为使用 [AMP 组件](/zh_cn/docs/reference/components.html)。在下一部分中，我们将探索我们的第一个 AMP 组件。
+{% endcall %}
+
+尝试打开外部 [`base.js`](https://github.com/googlecodelabs/accelerated-mobile-pages-foundations/blob/master/base.js) 文件。发现了什么？此文件应该不含任何 JavaScript 代码，并且只包含如下所示的信息注释：
+
+```javascript
+/*
+
+This external JavaScript file is intentionally empty.
+
+Its purpose is merely to demonstrate the AMP validation error related to the
+use of external JavaScript files.
+
+*/
+```
+
+由于此外部 JavaScript 文件并不是网站的功能性组件，因此我们可以安全地移除整个引用。
+
+从文档中**移除**以下外部 JavaScript 引用：
+
+```html
+<script type="text/javascript" src="base.js"></script>
+```
+
+现在，**重新加载**网页并验证脚本错误是否已消失。
+
+## 包含 AMP CSS 样板
+
+以下错误涉及缺失的样板代码：
+
+<pre class="error-text">
+必需的标记“noscript enclosure for boilerplate”缺失或不正确。
+必需的标记“head > style : boilerplate”缺失或不正确。
+必需的标记“noscript > style : boilerplate”缺失或不正确。
+</pre>
+
+每个 AMP 文档都必须具有以下 AMP 样板代码：
+
+```html
+<style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+```
+
+将该样板代码**添加**到文档的 `<head>` 标记底部。
+
+`<style amp-boilerplate>` 标记最初会隐藏正文内容，直到 AMP JavaScript 库加载完毕为止，然后正文内容才会呈现。AMP 之所以这样做，是为了防止系统呈现无样式内容，也称为无样式内容闪烁 (FOUC)。这样可确保网页的所有内容会同时显示，并且首屏中的所有内容都会一起呈现，从而实现具有真正即时感的用户体验。第二个标记可在浏览器停用 JavaScript 时还原此逻辑。
+
+## 将 `<img>` 替换为 `<amp-img>`
+
+AMP 不支持默认 HTML 对应项来显示媒体，这就解释了出现以下错误的原因：
+
+<pre class="error-text">
```

Comment:
All of these code blocks need to be in EN, so please update with EN translation (https://www.ampproject.org/docs/fundamentals/converting/resolving-errors).